### ISSUE TITLE
feature downsampling before Mamba

### ISSUE THREAD

**Issue by goutamyg** [Eyeline-Research/FlashDepth#5]
Hi @gene-chou 

Thank you for publishing the FlashDepth code and pretrained models!

In your implementation, the downsampling factor in the equation below is defined as 0.1 which means feature-maps are reduced to 1/10th of their spatial resolution. 
![Image](https://github.com/user-attachments/assets/5e13cfe9-68e2-4ddd-bf5e-fa9bda4e6493)
Doesn't it negatively impact the per-pixel supervision required to _align_ the depth at given frame with the predictions at previous frames? 

Have to experimentally verified the tradeoffs between multiple downsampling factors (say, 0.5, 0.25, 0.1) v/s corresponding accuracy and fps values?

-Goutam

**Comment by gene-chou** [comment]
Hey Goutam, we downsample the feature maps for speedup. Our assumption in the paper is that single-image depth models are already accurate, so we only need a "global" alignment, and in turn, a significantly downsampled feature map. 

Of course this assumption is not always true, but to improve consistency noticeably, one would need to experiment with the downsampling factors, the size of the mamba model, introduce temporal alignment / positional encoding...etc. 

We did run a grid search on the downsampling factors, but we observed drops in FPS without noticeable improvements in accuracy and consistency.

**Comment by goutamyg** [comment]
Thank you for your clarification! 

I am planning to re-train flashdepth under non-hybrid mode as shown below, to further reduce computational complexity. (mostly with ViT-small, i.e., depthanythingv2-small as the encoder). 

![Image](https://github.com/user-attachments/assets/da6f0668-372c-499a-b205-b691b31d1f67)

In this scenario, what changes are needed to get temporally stable results according to you? Do you think higher dhownsapming factor (say 0.5) would help in this case?

**Comment by gene-chou** [comment]
I think you can definitely try that. Maybe just don't downsample at all (factor=1.0).
However, you should probably add positional encoding. From frame to frame, the pixels will be moving / misaligned, so you should account for that. I don't have a good idea of how you would do the positional encoding without optical flow information though.
